akka-docs/rst/scala/code/docs/http/scaladsl/HttpServerExampleSpec.scala

```diff
@@ -380,4 +389,118 @@ class HttpServerExampleSpec extends WordSpec with Matchers {
       }
     }
   }
+
+  "stream random numbers" in compileOnlySpec {
+    //#stream-random-numbers
+    import akka.stream.scaladsl._
+    import akka.util.ByteString
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.model.{HttpEntity, ContentTypes}
+    import akka.http.scaladsl.server.Directives._
+    import akka.stream.ActorMaterializer
+    import scala.util.Random
+    import scala.io.StdIn
+
+    object WebServer {
+
+      def main(args: Array[String]) {
+
+        implicit val system = ActorSystem()
+        implicit val materializer = ActorMaterializer()
+        implicit val ec = system.dispatcher
+
+        // streams are re-usable so we can define it here
+        // and use it for every request
+        val numbers = Source.fromIterator(() =>
+          Iterator.continually(Random.nextInt()))
+
+        val route =
+          path("random") {
+            get {
+              complete(
+                HttpEntity(
+                  ContentTypes.`text/plain(UTF-8)`,
+                  // transform each number to a chunk of bytes
+                  numbers.map(n => ByteString(s"$n\n"))
+                )
+              )
+            }
+          }
+
+        val bindingFuture = Http().bindAndHandle(route, "localhost", 8080)
+        println(s"Server online at http://localhost:8080/\nPress RETURN to stop...")
+        StdIn.readLine() // let it run until user presses return
+        bindingFuture
+          .flatMap(_.unbind()) // trigger unbinding from the port
+          .onComplete(_ ⇒ system.terminate()) // and shutdown when done
+      }
+    }
+    //#stream-random-numbers
+  }
+
+
+  object Auction {
+    def props: Props = ???
+  }
+
+  "interact with an actor" in compileOnlySpec {
+    //#actor-interaction
+    import scala.concurrent.duration._
+    import akka.util.Timeout
+    import akka.pattern.ask
+    import akka.actor.ActorSystem
+    import akka.stream.ActorMaterializer
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.server.Directives._
+    import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+    import spray.json.DefaultJsonProtocol._
+    import scala.io.StdIn
+
+    object WebServer {
+
+      case class Bid(userId: String, bid: Int)
+      case object GetBids
+      case class Bids(bids: List[Bid])
+
+      implicit val bidFormat = jsonFormat2(Bid)
+      implicit val bidsFormat = jsonFormat1(Bids)
+
+      def main(args: Array[String]) {
+        implicit val system = ActorSystem()
+        implicit val materializer = ActorMaterializer()
+        implicit val ec = system.dispatcher
+
+        val auction = system.actorOf(Auction.props, "auction")
+
+        val route =
+          path("auction") {
+            put {
+              parameter("bid".as[Int], "user".as[String]) { (bid, user) =>
+                // place a bid, fire-and-forget
+                auction ! Bid(user, bid)
```

Comment:
this actually reminds me of a very common pattern that people like to implement... I think it's too complex for this doc but maybe we should do it in a template later on.
It's about `request(id) => map.getOrElseNewActor(id) ? bla`